internal/commands/accesslist/delete.go

```diff
@@ -0,0 +1,190 @@
+		cli.ProjectFlag(&cmd.inputs.Project),
+		cli.ProductFlag(&cmd.inputs.Products),
+		flags.StringSliceFlag{
+			Value: &cmd.inputs.Addresses,
+			Meta: flags.Meta{
+				Name: "ip",
+				Usage: flags.Usage{
+					Description: "Specify the IP address(es) or CIDR block(s) to delete",
+				},
+			},
+		},
+	}
+}
+
+// Inputs is the command inputs
+func (cmd *CommandDelete) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Handler is the command handler
+func (cmd *CommandDelete) Handler(profile *user.Profile, ui terminal.UI, clients cli.Clients) error {
+	app, err := cli.ResolveApp(ui, clients.Realm, cmd.inputs.Filter())
+	if err != nil {
+		return err
+	}
+
+	allowedIPs, err := clients.Realm.AllowedIPs(app.GroupID, app.ID)
+	if err != nil {
+		return err
+	}
+
+	selectedAllowedIPs, err := cmd.inputs.resolveAllowedIP(ui, allowedIPs)
+	if err != nil {
+		return err
+	}
+
+	if len(selectedAllowedIPs) == 0 {
+		return errors.New("No IP addresses or CIDR blocks to delete")
+	}
+
+	outputs := make(deleteAllowedIPOutputs, len(selectedAllowedIPs))
+	for i, allowedIP := range selectedAllowedIPs {
+		err := clients.Realm.AllowedIPDelete(app.GroupID, app.ID, allowedIP.ID)
+		outputs[i] = deleteAllowedIPOutput{allowedIP, err}
+	}
+
+	tableRows := make([]map[string]interface{}, 0, len(outputs))
+	for _, output := range outputs {
+		row := map[string]interface{}{
+			headerAddress: output.allowedIP.Address,
+			headerComment: output.allowedIP.Comment,
+		}
+		deleted := false
+		if output.err != nil {
+			row[headerDetails] = output.err
+		} else {
+			deleted = true
+		}
+		row[headerDeleted] = deleted
+		tableRows = append(tableRows, row)
+	}
+
+	ui.Print(terminal.NewTableLog(
+		fmt.Sprintf("Deleted %d IP address(es) and/or CIDR block(s)", len(outputs)),
+		deleteTableHeaders,
+		tableRows...,
+	))
+
+	return nil
+}
+
+func (i *deleteInputs) Resolve(profile *user.Profile, ui terminal.UI) error {
+	if err := i.ProjectInputs.Resolve(ui, profile.WorkingDirectory, false); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (i *deleteInputs) resolveAllowedIP(ui terminal.UI, allowedIPs []realm.AllowedIP) ([]realm.AllowedIP, error) {
+	if len(allowedIPs) == 0 {
+		return nil, nil
+	}
+
+	if len(i.Addresses) > 0 {
+		allowedIPsByAddress := make(map[string]realm.AllowedIP, len(allowedIPs))
+		for _, allowedIP := range allowedIPs {
+			allowedIPsByAddress[allowedIP.Address] = allowedIP
+		}
+
+		allowedIPs := make([]realm.AllowedIP, 0, len(i.Addresses))
+		notFound := false
+		var notFoundList []string
+		for _, identifier := range i.Addresses {
+			if allowedIP, ok := allowedIPsByAddress[identifier]; ok {
+				allowedIPs = append(allowedIPs, allowedIP)
+			} else {
+				notFound = true
+				notFoundList = append(notFoundList, identifier)
+			}
+		}
+
+		if notFound {
+			err := "unable to find IP address(es) and/or CIDR block(s): " + strings.Join(notFoundList, ", ")
+			return nil, errors.New(err)
```

Comment:
I'd just inline all of this, if you're worried about line length then something like this would be fine:

```go
return nil, errors.New(
  "unable to find IP address(es) and/or CIDR block(s): " +
    strings.Join(notFoundAddresses, ", "),
)
```